pkg/crypto/tinkcrypto/primitive/aead/subtle/aes_cbc.go

```diff
@@ -0,0 +1,154 @@
+		return nil, fmt.Errorf("aes_cbc: NewAESCBC() %w", err)
+	}
+
+	return &AESCBC{Key: key}, nil
+}
+
+// Encrypt encrypts plaintext using AES in CTR mode.
+// The resulting ciphertext consists of two parts:
+// (1) the IV used for encryption and (2) the actual ciphertext.
+func (a *AESCBC) Encrypt(plaintext []byte) ([]byte, error) {
+	plainTextSize := len(plaintext)
+	if plainTextSize > maxInt-AESCBCIVSize {
+		return nil, errors.New("aes_cbc: plaintext too long")
+	}
+
+	iv := a.newIV()
+
+	cbc, err := newCipher(a.Key, iv, false)
+	if err != nil {
+		return nil, fmt.Errorf("aes_cbc: Encrypt() %w", err)
+	}
+
+	ciphertext := make([]byte, AESCBCIVSize+plainTextSize)
+	if n := copy(ciphertext, iv); n != AESCBCIVSize {
+		return nil, fmt.Errorf("aes_cbc: failed to copy IV (copied %d/%d bytes)", n, AESCBCIVSize)
+	}
+
+	if n := copy(ciphertext[AESCBCIVSize:], plaintext); n != plainTextSize {
+		return nil, fmt.Errorf("aes_cbc: failed to copy plaintext (copied %d/%d bytes)", n, plainTextSize)
+	}
+
+	ciphertext = Pad(ciphertext, plainTextSize, cbc.BlockSize())
+
+	cbc.CryptBlocks(ciphertext[AESCBCIVSize:], ciphertext[AESCBCIVSize:])
+
+	return ciphertext, nil
+}
+
+// Decrypt decrypts ciphertext.
+func (a *AESCBC) Decrypt(ciphertext []byte) ([]byte, error) {
+	ciphertextSize := len(ciphertext)
+	if ciphertextSize < AESCBCIVSize {
+		return nil, errors.New("aes_cbc: ciphertext too short")
+	}
+
+	iv := ciphertext[:AESCBCIVSize]
+
+	cbc, err := newCipher(a.Key, iv, true)
+	if err != nil {
+		return nil, fmt.Errorf("aes_cbc: Decrypt() %w", err)
+	}
+
+	blockSize := cbc.BlockSize()
+
+	if len(ciphertext[AESCBCIVSize:])%blockSize > 0 {
+		return nil, errors.New("aes_cbc: invalid ciphertext padding")
+	}
+
+	plaintext := make([]byte, ciphertextSize-AESCBCIVSize)
+	cbc.CryptBlocks(plaintext, ciphertext[AESCBCIVSize:])
+
+	if len(plaintext) == 0 {
+		return plaintext, nil
+	}
+
+	// unpad plaintext if not block size.
+	last := plaintext[len(plaintext)-1]
+	count := int(last)
+
+	if count == 0 || count > blockSize || count > len(plaintext) {
+		return nil, errors.New("aes_cbc: invalid padding")
+	}
+
+	padding := bytes.Repeat([]byte{last}, count)
+	if !bytes.HasSuffix(plaintext, padding) { // padding not found, return full plaintext.
+		return plaintext, nil
+	}
+
+	// padding was found, trim it and return remaining plaintext.
+	return plaintext[:len(plaintext)-len(padding)], nil
+}
+
+// newIV creates a new IV for encryption.
+func (a *AESCBC) newIV() []byte {
+	return random.GetRandomBytes(uint32(AESCBCIVSize))
+}
+
+// newCipher creates a new AES-CBC cipher using the given key, IV and the crypto library.
+func newCipher(key, iv []byte, decrypt bool) (cipher.BlockMode, error) {
+	block, err := aes.NewCipher(key)
+	if err != nil {
+		return nil, fmt.Errorf("aes_cbc: failed to create block cipher, error: %w", err)
+	}
+
+	// If the IV is less than BlockSize bytes we need to pad it with zeros otherwise NewCBCEncrypter will panic.
+	if len(iv) < aes.BlockSize {
```

Comment:
makes sense since  we [generate](https://github.com/hyperledger/aries-framework-go/pull/2811/files/9d05bdf574fdad1d59b54e5343e266b5f4ed7ae6#diff-108bd5df34a666a2a61133481aa53ee51e671debf7037f79de5a048f4e74bcb4R47) a new iv and pass it to newCipher(), iv will never be < BlockSize

I'll remove this padding logic and return an error instead